### PR TITLE
[KOA-5123] Example for tokens

### PR DESCRIPTION
**Pull request by frugoman**
KOA-5123

Added example screens for Colors, Spacing and Radii for UIKit

### CODE REVIEW

File: Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+            verticalStackView.bottomAnchor.constraint(equalTo: scrollView.bottomAnchor, constant: -BPKSpacingLg),
+            verticalStackView.leadingAnchor.constraint(equalTo: scrollView.leadingAnchor, constant: BPKSpacingLg),
+            verticalStackView.trailingAnchor.constraint(equalTo: scrollView.trailingAnchor, constant: BPKSpacingLg),
+            verticalStackView.widthAnchor.constraint(equalTo: scrollView.widthAnchor)
+        ])
+        
+        colorTokens.tokens
+            .map(colorSection(section:))
+            .forEach(verticalStackView.addArrangedSubview)
+    }
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:102]: Can we move this to a setup function?

File: Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        NSLayoutConstraint.activate([
+            colorView.leadingAnchor.constraint(equalTo: container.leadingAnchor, constant: BPKSpacingMd),
+            colorView.widthAnchor.constraint(equalToConstant: BPKSpacingLg),
+            colorView.heightAnchor.constraint(equalToConstant: BPKSpacingLg),
+            colorView.centerYAnchor.constraint(equalTo: container.centerYAnchor),
+            
+            colorLabel.topAnchor.constraint(equalTo: container.topAnchor, constant: BPKSpacingMd),
+            colorLabel.leadingAnchor.constraint(equalTo: colorView.trailingAnchor, constant: BPKSpacingMd),
+            colorLabel.bottomAnchor.constraint(equalTo: container.bottomAnchor, constant: -BPKSpacingMd),
+            colorLabel.trailingAnchor.constraint(equalTo: container.trailingAnchor, constant: -BPKSpacingMd)
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:149]: Can we 'flip this around'? So we don't need the `-` on the spacing?

So container.trailingAnchor.constraint(equalTo: colorLabel.trailingAnchor, constant: <spacing>)

File: Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        struct Section {
+            let name: String
+            let colors: [Color]
+            
+            // swiftlint:disable:next nesting
+            struct Color {
+                let color: UIColor
+                let name: String
+            }
+        }
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:172]: Does this have to be nested? I don't like the swiftlint:disable calls here.

File: Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        radiiViews.forEach(verticalStackView.addArrangedSubview)
+        view.addSubview(verticalStackView)
+        let safeArea = view.safeAreaLayoutGuide
+        NSLayoutConstraint.activate([
+            verticalStackView.topAnchor.constraint(equalTo: safeArea.topAnchor, constant: BPKSpacingLg),
+            verticalStackView.leadingAnchor.constraint(equalTo: safeArea.leadingAnchor, constant: BPKSpacingLg),
+            verticalStackView.trailingAnchor.constraint(equalTo: safeArea.trailingAnchor, constant: -BPKSpacingLg),
+            verticalStackView.bottomAnchor.constraint(lessThanOrEqualTo: safeArea.bottomAnchor, constant: -BPKSpacingLg)
+        ])
+
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift:54]: Same here, can we have a setup function?

File: Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        label.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(label)
+        NSLayoutConstraint.activate([
+            label.topAnchor.constraint(equalTo: view.topAnchor, constant: BPKSpacingXl),
+            label.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: BPKSpacingLg),
+            label.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -BPKSpacingLg),
+            label.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -BPKSpacingXl)
+        ])
+
+        return view
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift:74]: Can we add some spacing to divide the code into ''blocks'? Same comment applies to the negative spacing (as above)

File: Example/Backpack/UIKit/Tokens/SpacingTokensViewController.swift [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        let spacingViews = spacingValues.map(spacingView(for:))
+        spacingViews.forEach(verticalStackView.addArrangedSubview)
+        view.addSubview(verticalStackView)
+        let safeArea = view.safeAreaLayoutGuide
+        NSLayoutConstraint.activate([
+            verticalStackView.topAnchor.constraint(equalTo: safeArea.topAnchor, constant: BPKSpacingLg),
+            verticalStackView.leadingAnchor.constraint(equalTo: safeArea.leadingAnchor, constant: BPKSpacingLg),
+            verticalStackView.trailingAnchor.constraint(equalTo: safeArea.trailingAnchor, constant: -BPKSpacingLg),
+            verticalStackView.bottomAnchor.constraint(lessThanOrEqualTo: safeArea.bottomAnchor, constant: -BPKSpacingLg)
+        ])
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/SpacingTokensViewController.swift:56]: Same here

**Review by gert-janvercauteren — COMMENTED** [review]
Just some minor comments around code structure

File: Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift (hunk 3 above)
Comment (frugoman) [reply Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:172]: dont like the :disable either. Nesting I do think is fine, as Section and Color are structs that have no use outside that scope, i'll move to extensions

Comment (frugoman) [reply Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:172]: not fan of that rule really, as I think nesting is quite useful to scope properly, but anyways....

Comment (gert-janvercauteren) [reply Example/Backpack/UIKit/Tokens/ColorTokensViewController.swift:172]: We can keep it nested, should we reconsider this rule then?

File: Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        displayTokens()
+    }
+    
+    private func displayTokens() {
+        radiiValues.map(radiiView(for:))
+            .forEach(verticalStackView.addArrangedSubview)
+    }
+    
+    private func setupView() {
+        view.backgroundColor = .systemBackground
```
Comment (gert-janvercauteren) [inline Example/Backpack/UIKit/Tokens/RadiusTokensViewController.swift:54]: Can we use canvasColor?

**Review by gert-janvercauteren — APPROVED** [review]

**Pull request merged**